runtime/src/drivers.rs

```diff
@@ -582,6 +582,22 @@ impl Drivers {
         }
     }
 
+    /// Checks if the caller is privilege level 0
+    ///
+    /// # Arguments
+    ///
+    /// * `pl0_pauser` - Value of PL0 PAuser
+    /// * `flags` - Flags from manifest header
+    /// * `locality` - Caller's locality
+    pub fn is_caller_pl0(pl0_pauser: u32, flags: u32, locality: u32) -> bool {
```

Comment:
If you make this a member function on the drivers struct (`pub fn is_caller_pl0(&self, ...)`) are you able to remove all the parameters? I think all of this information comes from various drivers. So we can probably hide all these details from the caller.

Reply:
I was able to make the public interface a member function. The refactor also bubbled up to `is_dpe_context_threshold_exceeded` since it calls `is_caller_pl1`. I went ahead and also refactored this method, but because of some logic in `initialize_dpe` I had to keep a private non member function.

There may be a better way, but I didn't find an obvious way to de-couple the newly created DPE instance from the driver, as said instance is holding a mutable reference to the driver by the time the second call to `DeriveContextCmd` happens. 

To work around this I kept a private helper method for `is_dpe_context_threshold_exceeded`.

Reply:
Looks good, thanks!